packages/protocol/test-sol/stability/PairManager.t.sol

```diff
@@ -0,0 +1,431 @@
+      abi.encodeWithSelector(mockReserve.isCollateralAsset.selector, address(testCollateralAsset)),
+      abi.encode(true)
+    );
+
+    PairManager.Pair memory newPair;
+    newPair.stableAsset = address(testStableAsset);
+    newPair.collateralAsset = address(testCollateralAsset);
+    newPair.mentoExchange = IMentoExchange(address(testMentoExchange));
+    newPair.collateralBucketFraction = FixidityLib.wrap(0.1 * 10**24);
+    newPair.stableBucketMaxFraction = FixidityLib.wrap(0.1 * 10**24);
+    newPair.spread = FixidityLib.wrap(0.1 * 10**24);
+
+    testee.createPair(newPair);
+  }
+}
+
+contract PairManagerTest_initilizerSettersGetters is PairManagerTest {
+  /* ---------- Initilizer ---------- */
+
+  function test_initilize_shouldSetOwner() public view {
+    assert(testee.owner() == deployer);
+  }
+
+  function test_initilize_shouldSetBroker() public view {
+    assert(testee.broker() == broker);
+  }
+
+  /* ---------- Setters ---------- */
+
+  function test_setBroker_whenSenderIsNotOwner_shouldRevert() public {
+    changePrank(notDeployer);
+    vm.expectRevert("Ownable: caller is not the owner");
+    testee.setBroker(address(0));
+  }
+
+  function test_setBroker_whenAddressIsZero_shouldRevert() public {
+    vm.expectRevert("Broker address must be set");
+    testee.setBroker(address(0));
+  }
+
+  function test_setBroker_whenSenderIsOwner_shouldUpdateAndEmit() public {
+    address newBroker = actor("newBroker");
+    vm.expectEmit(true, false, false, false);
+    emit BrokerUpdated(newBroker);
+
+    testee.setBroker(newBroker);
+
+    assert(testee.broker() == newBroker);
+  }
+
+  function test_updateBuckets_whenSenderIsNotBroker_shouldRevert() public {
+    bytes32 pairId = keccak256(
+      abi.encodePacked(
+        testStableAsset.symbol(),
+        testCollateralAsset.symbol(),
+        testMentoExchange.name()
+      )
+    );
+
+    vm.expectRevert("Caller is not the Broker");
+    testee.updateBuckets(pairId, 45000000000, 45000000000);
+  }
+
+  function test_updateBuckets_whenPairDoesNotExist_shouldRevert() public {
+    changePrank(broker);
+
+    bytes32 pairId = keccak256(
+      abi.encodePacked(
+        anotherTestStableAsset.symbol(),
+        testCollateralAsset.symbol(),
+        testMentoExchange.name()
+      )
+    );
+
+    vm.expectRevert("A pair with the specified id does not exist");
+    testee.updateBuckets(pairId, 45000000000, 45000000000);
+  }
+
+  function test_updateBuckets_whenSenderIsBroker_shouldUpdateAndEmit() public {
+    changePrank(broker);
+
+    bytes32 pairId = keccak256(
+      abi.encodePacked(
+        testStableAsset.symbol(),
+        testCollateralAsset.symbol(),
+        testMentoExchange.name()
+      )
+    );
+
+    PairManager.Pair memory pairBeforeUpdate = testee.getPair(pairId);
+    assert(pairBeforeUpdate.stableBucket == 250000000000);
+    assert(pairBeforeUpdate.collateralBucket == 250000000000);
+
+    vm.expectEmit(false, false, false, false);
```

Comment:
Same as above, I think all false means that this matches `BucketsUpdated` with any arguments, right?
I usually just go for `true, true, true, true` and dial back when needed.